Packet++/header/S7commLayer.h

```diff
@@ -0,0 +1,234 @@
+#ifndef PCAPPLUSPLUS_S7COMMLAYER_H
+#define PCAPPLUSPLUS_S7COMMLAYER_H
+
+#include "EthLayer.h"
+#include "Layer.h"
+
+namespace pcpp
+{
+/**
+ * @struct s7commhdr
+ * Represents a S7COMM protocol header
+ */
+#pragma pack(push, 1)
+	typedef struct
+	{
+		/** protocol id */
+		uint8_t protocol_id;
+		/** message type */
+		uint8_t msg_type;
+		/** redundancy identification (reserved) */
+		uint16_t reserved;
+		/** protocol data unit reference */
+		uint16_t pdu_ref;
+		/** parameter length */
+		uint16_t param_length;
+		/** data length */
+		uint16_t data_length;
+	} s7commhdr;
+#pragma pack(pop)
+
+/**
+ * @struct s7comm_ack_data_hdr
+ * Represents a S7COMM protocol with Ack-Data header
+ */
+#pragma pack(push, 1)
+	struct s7comm_ack_data_hdr : s7commhdr
+	{
+		/** error class */
+		uint8_t error_class;
+		/** error code */
+		uint8_t error_code;
+	};
+#pragma pack(pop)
+
+	/**
+	 * @class S7CommParameter
+	 * Represents a S7COMM (S7 Communication7) protocol Parameter
+	 */
+	class S7CommParameter
+	{
+		friend class S7commLayer;
+
+	  public:
+		S7CommParameter() {}
+
+		virtual ~S7CommParameter() {}
+
+		/**
+		 * @return The data of the Parameter
+		 */
+		uint8_t *getData() const { return m_Data; }
+		/**
+		 * @return The length of the Parameter data
+		 */
+		size_t getDataLength() const { return m_DataLen; }
+
+	  private:
+		S7CommParameter(uint8_t *data, size_t dataLen) : m_Data(data), m_DataLen(dataLen) {}
+		uint8_t *m_Data;
+		size_t m_DataLen;
+	};
+	/**
+	 * @class S7commLayer
+	 * Represents a S7COMM (S7 Communication) protocol
+	 */
+	class S7commLayer : public Layer
+	{
+	  public:
+		/**
+		 * A constructor that allocates a new S7comm header
+		 * @param[in] msg_type The general type of the message
+		 * @param[in] pdu_ref Link responses to their requests
+		 * @param[in] param_length The length of the parameter field
+		 * @param[in] data_length The length of the data field
+		 * @param[in] error_class The value of the error class
+		 * @param[in] error_code The value of the error code
+		 */
+		S7commLayer(uint8_t msg_type, uint16_t pdu_ref, uint16_t param_length, uint16_t data_length,
```

Comment:
Let's update all argument names, both in this method and in the other methods of this class from `kebab_case` to `camelCase` (for example: `msgType` instead of `msg_type`)?